### PR TITLE
STYLE:clangFormat pre-commit hook

### PR DESCRIPTION
**Pull request by ViktorvdValk**
- A pre-commit hook for local git usage on both Windows and Unix systems.
- Pre-commit hook use update in CONTRIBUTING.md

### CODE REVIEW

**Comment by ViktorvdValk** [comment]
This would close #329

File: .github/workflows/ElastixGitHubActions.yml [hunk 1]
```diff
@@ -18,20 +18,20 @@ jobs:
           - os: ubuntu-18.04
             c-compiler: "gcc"
             cxx-compiler: "g++"
-            itk-git-tag: "v5.1.1"
```
Comment (N-Dekker) [inline .github/workflows/ElastixGitHubActions.yml:21]: Should this ITK upgrade not be done in a separate commit or PR? Or is it related to clang-format?

Comment (ViktorvdValk) [reply .github/workflows/ElastixGitHubActions.yml:21]: Yes, this was already done in commit 98aaaf7e27c7dacd271db5a2cc823b938a154392, so I don't get why it still shows up here.
But no it's not related to clang-format.

Comment (N-Dekker) [reply .github/workflows/ElastixGitHubActions.yml:21]: git happens 😸

**Comment by mstaring** [comment]
I think a rebase is needed

**Comment by ViktorvdValk** [comment]
I included the clang-format binaries in the repo now, which is not really conventional.
However the other option is downloading them at a git clone,
but than you rely on an external website, which I found trickier than just including them,
since they're not supposed to be updated anymore.

**Review by N-Dekker — CHANGES_REQUESTED** [review]
Sorry, I'm still not so sure if it's OK to store those binaries in the elastix git repo. Especially because they may need to be updated now and then, which would quickly make an elastix clone bigger and bigger. Did you also have a look at how ITK did it?

**Comment by ViktorvdValk** [comment]
@N-Dekker now the binaries are downloaded from kitware first in the bash script, so they're not stored on github anymore.

**Comment by N-Dekker** [comment]
> now the binaries are downloaded from kitware first in the bash script, so they're not stored on github anymore.

Cool, I'm gonna give it a try!

**Review by N-Dekker — APPROVED** [review]
If possibly, could you still figure out how to print a message, saying something like "Info: clang-format made some changes after the commit", whenever clang-format has actually modified the source code through this hook?

Otherwise, just merge! I like it very much :-)
